File: 1442-count-triplets-that-can-form-two-arrays-of-equal-xor/1442-count-triplets-that-can-form-two-arrays-of-equal-xor.py
class Solution:
    def countTriplets(self, arr: List[int]) -> int:
        n = len(arr)
        res = 0
        for i in range(n):
            pre = arr[i]
            for j in range(i + 1, n):
                pre ^= arr[j]
                res += (j - i) if(pre == 0) else 0
        
        return res

File: 1442-count-triplets-that-can-form-two-arrays-of-equal-xor/1442-count-triplets-that-can-form-two-arrays-of-equal-xor.go
func countTriplets(arr []int) int {
    var xorArr = []int{}
    xorArr = append(xorArr, arr[0])
    for _, a := range arr[1:] {
        xorArr = append(xorArr, xorArr[len(xorArr) - 1] ^ a)
    }
    
    var counts = 0
    for i := range len(arr) {
        var subPrev = 0
        if i > 0 {
            subPrev = xorArr[i - 1]
        }
        
        for j := i + 1; j < len(arr); j++ {
            if subPrev ^ xorArr[j] == 0 {
                counts += (j - i)
            }
        }
    }
    
    return counts
}